nautobot/extras/models/groups.py

```diff
@@ -494,14 +494,37 @@ def generate_query_for_filter(self, filter_field, value):
             Value passed to the filter
         """
         query = models.Q()
+        field_name = filter_field.field_name
+
+        # Attempt to account for `ModelChoiceFilter` where `to_field_name` MAY be set.
+        to_field_name = getattr(filter_field.field, "to_field_name", None)
+        if to_field_name is not None:
+            field_name = f"{field_name}__{to_field_name}"
+
+        lookup = f"{field_name}__{filter_field.lookup_expr}"
 
         # Explicitly call generate_query_{filter_method} for a method filter.
         if filter_field.method is not None and hasattr(filter_field.parent, "generate_query_" + filter_field.method):
             filter_method = getattr(filter_field.parent, "generate_query_" + filter_field.method)
-            query |= models.Q(filter_method(value))
+            query |= filter_method(value)
+
+        # Explicitly call `filter_field.generate_query` for a reversible filter.
+        elif hasattr(filter_field, "generate_query"):
+            # Is this a list of strings? Well let's resolve it to related model objects so we can
+            # pass it to `generate_query` to get a correct Q object back out. When values are being
+            # reconstructed from saved filters, lists of slugs are common e.g. (`{"site": ["ams01",
+            # "ams02"]}`, the value being a list of site slugs (`["ams01", "ams02"]`).
+            if value and isinstance(value, list) and isinstance(value[0], str):
```

Comment:
Any chance that `value` may be a list but of length 0?

Reply:
Probably, but if that were the case an empty list would evaluate to `False` and wouldn't be in scope.